### PR TITLE
Remove Android hacks (fixes #2505)

### LINKED ISSUE #2505: Clean up old Android DNS hacks
Per #2290 we no longer need hacks for the usage reporting server, and possibly some other things.

### PR DESCRIPTION
**Pull request by calmh**
Also fixes what I think migh thave been a bug where we did not use the
proxy for usage reports. And removes the BuildEnv field that we don't
need any more.

### CODE REVIEW

File: lib/config/config.go [hunk 1]
```diff
@@ -50,7 +50,6 @@ var (
 	DefaultDiscoveryServers = append(DefaultDiscoveryServersV4, DefaultDiscoveryServersV6...)
 
 	// DefaultDiscoveryServersIP is used by the usage reporting.
-	// XXX: Detect Android, and use this is we still don't have working DNS?
 	DefaultDiscoveryServersIP = []string{
```
Comment (AudriusButkevicius) [inline lib/config/config.go]: Why do we still need this?

Comment (calmh) [inline lib/config/config.go]: Things may be configured to use the IP addresses, and it's neat if we detect that case for the usage reporting? We should remove it at some point though...

Comment (AudriusButkevicius) [inline lib/config/config.go]: If DNS works, just nuke it.

Comment (calmh) [inline lib/config/config.go]: Fair nuff :)

Comment (AudriusButkevicius) [inline lib/config/config.go]: Let's wait for it to get out of beta though.

**Comment by calmh** [comment]
I haven't heard any screaming so I think the Android stuff is good now and this could be merged?

**Comment by Zillode** [comment]
:+1:

**Pull request merged**